packages/data-service/src/data-service.ts

```diff
@@ -118,6 +128,27 @@ interface CompassClientSession extends ClientSession {
   [kSessionClientType]: ClientType;
 }
 
+// In nodejs, AbortController support is from 15.x and we are currently on 14.x.
+// With the implementation of cancellable actions using AbortSignal, we have custom AbortSignal
+// type definition to avoid including DOM compiler options in tsconfig.
+type AbortSignal = {
+  aborted: boolean;
+  onabort: ((this: AbortSignal, event: Event) => void) | null;
+  addEventListener: (
+    type: string,
+    listener: (event: Event) => void,
+    options: Record<string, unknown>
+  ) => void;
+  removeEventListener: (type: string, listener: (event: Event) => void) => void;
+  dispatchEvent: (event: Event) => boolean;
```

Comment:
Where is `Event` included from? If it’s the DOM `Event` class, then we probably already require the DOM type definitions here.